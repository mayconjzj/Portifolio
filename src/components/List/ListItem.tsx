export const ListItem = ({ children }: { children: React.ReactNode }) => {
  return <li className="list-none">{children}</li>;
};
