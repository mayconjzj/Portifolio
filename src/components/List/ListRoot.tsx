import { twMerge } from 'tailwind-merge';

import { ListRootProps } from '@/models/List';

export const ListRoot = ({ children, ...rest }: ListRootProps) => {
  return (
    <ul className={twMerge('flex gap-7', rest.className)} {...rest}>
      {children}
    </ul>
  );
};
