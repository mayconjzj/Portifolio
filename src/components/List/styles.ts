import tw from 'tailwind-styled-components';

export const List = tw.ul`
  flex gap-x-7
`;
