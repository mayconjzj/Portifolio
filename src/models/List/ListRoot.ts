export type ListRootProps = {
  children: React.ReactNode;
  className?: string;
};
