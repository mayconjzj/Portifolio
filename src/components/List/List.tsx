import * as S from './styles';

export const List = ({ children }: { children: React.ReactNode }) => {
  return <S.List>{children}</S.List>;
};
