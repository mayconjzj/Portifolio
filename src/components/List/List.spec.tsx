import { render } from '@testing-library/react';

import { List } from './List';

describe('List', () => {
  it('should render the list', () => {
    const { container } = render(<List>Next test</List>);
    expect(container).toMatchSnapshot();
  });
});
