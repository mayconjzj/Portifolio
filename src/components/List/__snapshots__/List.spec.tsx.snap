// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the list 1`] = `
<DocumentFragment>
  <ul
    class="flex gap-x-7"
  >
    TESTE
  </ul>
</DocumentFragment>
`;
